sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -270,6 +271,19 @@ class SessionCatalog(
   }
 
   /**
+   * Retrieve the metadata of an existing permanent table/view. If no database is specified,
+   * assume the table/view is in the current database. If the specified table/view is not found
+   * in the database then a [[NoSuchTableException]] is thrown.
+   */
+  def getNonTempTableMetadata(name: TableIdentifier): CatalogTable = {
```

Comment:
In your PR https://github.com/apache/spark/pull/14962, `getTableMetadata` will not deal with temporary table. To simplify the potential conflict resolution, I just introduce a new function `getNonTempTableMetadata`. Thus, you can just basically replace it by the new `getTableMetadata`. Does it sound OK to you? Sorry for any inconvience it might cause.